### ISSUE TITLE
Generating many-to-many relationship with itself

### ISSUE THREAD

**Issue by WhyINeedToFillUsername** [jhipster/jhipster-uml#184]
### The issue
I have an entity Flat and I want to add a relationship with itself (Flat has friends). It is bidirectional, when Flat "A" has a friend "B", the "B" has friend "A". This leads me to Many-To-Many relationship.

The problem is when I cannot specify it in the JDL, I have to do something like:
```
relationship ManyToMany {
	Flat{friends} to Flat{friends}
}
```
which generates
```
    {
      "relationshipId": 3,
      "relationshipType": "many-to-many",
      "relationshipName": "friends",
      "otherEntityName": "flat",
      "otherEntityField": "id",
      "ownerSide": true
    },
    {
      "relationshipId": 4,
      "relationshipType": "many-to-many",
      "relationshipName": "friends",
      "otherEntityName": "flat",
      "ownerSide": false,
      "otherEntityRelationshipName": "friends"
    }
```
This creates liquibase changelog:
```
<addForeignKeyConstraint baseColumnNames="flats_id"
                                 baseTableName="flat_friend_of"
                                 constraintName="fk_flat_friend_of_flat_id"
                                 referencedColumnNames="id"
                                 referencedTableName="flat"/>
        <addForeignKeyConstraint baseColumnNames="friend_ofs_id"
                                 baseTableName="flat_friend_of"
                                 constraintName="fk_flat_friend_of_flat_id"
                                 referencedColumnNames="id"
                                 referencedTableName="flat"/>
```
which has same name for `constraintName` (that leads to error when running the app.) Though I believe the wrongly generated liquibase changelog is beyond the scope of the jhipster-uml.

It also creates duplicated fields in the `Flat.java` (as the name of the relationship is same):

 ```
 @ManyToMany
 @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
 @JoinTable(name = "flat_friends",
            joinColumns = @JoinColumn(name="flats_id", referencedColumnName="ID"),
            inverseJoinColumns = @JoinColumn(name="friends_id", referencedColumnName="ID"))
 private Set<Flat> friends = new HashSet<>();

 @ManyToMany(mappedBy = "friends")
 @JsonIgnore
 @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
 private Set<Flat> friends = new HashSet<>();
...
public Set<Flat> getFriends() {
     return friends;
 }

 public void setFriends(Set<Flat> flats) {
     this.friends = flats;
 }

 public Set<Flat> getFriends() {
     return friends;
 }

 public void setFriends(Set<Flat> flats) {
     this.friends = flats;
 }
```

To sum up, is there another "correct" way of managing this kind of relationship using the tools jhipster supplies?
There is at least the bug with the same name of `constraintName`.

### Versions

| JHipster           | JHipster UML           |
| ------------------ | ---------------------- |
| `3.1.0` |    `1.6.5.`   |

**Comment by MathieuAA** [comment]
Hi, to my knowledge: no. I thought I had a solution but didn't test it because I realized it was beyond my scope (but not beyond JHipster's)... Did you try with the questions and answers?

**Comment by WhyINeedToFillUsername** [comment]
As I designed the domain in the JDL, I did not try the q&a.

**Comment by MathieuAA** [comment]
Hi, this doesn't work as there's an error when `mvn test`-ing the relationships (with the Q&As):
```
[ERROR] /home/mathieu/jhipster/test/src/main/java/com/mycompany/myapp/domain/A.java:[51,10] cannot find symbol
[ERROR] symbol:   method getOtherAS()
[ERROR] location: variable a of type com.mycompany.myapp.domain.A
[ERROR] /home/mathieu/jhipster/test/src/main/java/com/mycompany/myapp/domain/A.java:[57,10] cannot find symbol
[ERROR] symbol:   method getOtherAS()
[ERROR] location: variable a of type com.mycompany.myapp.domain.A
```
You could post an issue to the generator's issue tracker and refer to this issue if you want this feature.
In the meantime, I'm closing this.
